stistools/splice.py

```diff
@@ -0,0 +1,888 @@
+
+    Parameters
+    ----------
+    array : ``numpy.array``
+        Target array.
+    target_value : ``float``
+        Target value.
+
+    Returns
+    -------
+    index : ``int``
+        Index of the value in ``array`` that is closest to ``target_value``.
+    """
+    index = array.searchsorted(target_value)
+    index = np.clip(index, 1, len(array) - 1)
+    left = array[index - 1]
+    right = array[index]
+    index -= target_value - left < right - target_value
+    return index
+
+
+# Read the Echelle spectrum based on dataset name and a prefix for file location
+def read_spectrum(x1d_input, truncate_edge_left=None, truncate_edge_right=None):
+    """
+    This is a fairly straightforward function to read the spectrum from a `x1d`
+    FITS file.
+
+    Parameters
+    ----------
+    x1d_input : ``str``
+        Path and name of the ``*_x1d.fits`` file containing the spectrum.
+
+    truncate_edge_left : ``int``, optional
+        Set the number of low-resolution pixels at the left edge of the detector
+        where the spectra should be truncated. If ``None``, then no truncation
+        is applied. Default is ``None``.
+
+    truncate_edge_right : ``int``, optional
+        Set the number of low-resolution pixels at the right edge of the
+        detector where the spectra should be truncated. If ``None``, then no
+        truncation is applied. Default is ``None``.
+
+    Returns
+    -------
+    spectrum : ``list``
+        List of all the orders contained in the Echelle spectrum and their
+        respective fluxes.
+    """
+    with fits.open(x1d_input) as hdu:
+        header = hdu[0].header
+        data = hdu['SCI'].data
+    optical_element = header['OPT_ELEM']
+    if optical_element[0] != 'E':
+        raise TypeError("This is not an Echelle spectrum.")
+    wavelength = data['WAVELENGTH']
+    flux = data['FLUX']
+    uncertainty = data['ERROR']
+    data_quality = data['DQ']
+    gross_counts = data['GROSS']
+    net_counts = data['NET']
+    n_orders = len(wavelength)
+
+    # We index the spectral regions as a `dict` in order to avoid confusion with
+    # too many numerical indexes. Also, since the orders are in reverse order,
+    # we index them in the opposite way
+    spectrum = [{'wavelength': wavelength[-i - 1],
+                 'flux': flux[-i - 1],
+                 'uncertainty': uncertainty[-i - 1],
+                 'data_quality': data_quality[-i - 1],
+                 'gross': gross_counts[-i - 1],
+                 'net': net_counts[-i - 1]}
+                for i in range(n_orders)]
+
+    # If edge truncation is passed, we simply assign a different DQ flags to the
+    # pixels to be truncated. We choose the flag 4096, which corresponds to
+    # "Extracted flux affected by bad input data."
+    if truncate_edge_right is not None:
+        for sk in spectrum:
+            sk['data_quality'][-truncate_edge_right:] = 4096
+    elif truncate_edge_left is not None:
```

Comment:
`elif` -> `if` allows both edges to be truncated with default values.  Is that the desired behavior?